tool/tsh/common/tsh.go

```diff
@@ -753,7 +770,8 @@ func Run(ctx context.Context, args []string, opts ...CliOption) error {
 	ssh.Flag("x11-untrusted-timeout", "Sets a timeout for untrusted X11 forwarding, after which the client will reject any forwarding requests from the server").Default("10m").DurationVar((&cf.X11ForwardingTimeout))
 	ssh.Flag("participant-req", "Displays a verbose list of required participants in a moderated session.").BoolVar(&cf.displayParticipantRequirements)
 	ssh.Flag("request-reason", "Reason for requesting access").StringVar(&cf.RequestReason)
-	ssh.Flag("disable-access-request", "Disable automatic resource access requests").BoolVar(&cf.disableAccessRequest)
+	ssh.Flag("request-mode", fmt.Sprintf("Type of automatic access request to make (%s)", strings.Join(accessRequestModes, ", "))).Default(accessRequestModeResource).EnumVar(&cf.RequestMode, accessRequestModes...)
```

Comment:
Are users going to be happy having to specify `--request-mode=role` every time they want to `tsh ssh` with the option of doing an automatic role request? I think it could be a big UX improvement to make it automatic if they can't do a resource request but can do a role request. When we made automatic resource requests, we really wanted to make it seamless without extra manual steps, I imagine we'd want the same here

Reply:
Makes sense, but I'm not sure how you'd get into a scenario where we can do a role request but not a resource request (in the context of `tsh ssh`). If you can do a role request, then you must have SearchAsRoles, so you can also do a resource request.

Reply:
yeah, I didn't realize search_as_roles were required at the time, disregard this then